cpp/src/arrow/compute/kernels/scalar_cast_string.cc

```diff
@@ -286,17 +288,20 @@ Status CastBinaryToBinaryOffsets<int64_t, int32_t>(KernelContext* ctx,
   }
 }
 
+// Span -> Span
```

Comment:
Why "span"? Did you mean "string"? Or perhaps "Non-view"?

Reply:
It's the term used in the rest of the code.

Reply:
I don't think it's used in this file, and span is probably too overloaded to use for this anywhere in arrow.
```suggestion
// Offset String -> Offset String
```